src/dioptra/restapi/v1/utils.py

```diff
@@ -62,16 +62,14 @@ def build_user(user: models.User) -> dict[str, Any]:
     }
 
 
-def build_current_user(user: models.Group) -> dict[str, Any]:
-    member_of = {x.group.group_id: x.group for x in user.group_memberships}
-    manager_of = {x.group.group_id: x.group for x in user.group_managerships}
-    groups = {**member_of, **manager_of}.values()
-
+def build_current_user(user: models.User) -> dict[str, Any]:
     return {
         "id": user.user_id,
         "username": user.username,
         "email": user.email_address,
-        "groups": [build_group_ref(group) for group in groups],
+        "groups": [
+            build_group_ref(membership.group) for membership in user.group_memberships
```

Comment:
same thought process here: we could miss a group that the user is associated with if it can be the case that they are a manager but not a member,